### DISCUSSION TITLE
sudo airmon-ng start wlan0  ls: cannot access '/sys/class/net': No such file or directory

### DISCUSSION THREAD

**Discussion by demetrebadzaradze** [aircrack-ng/aircrack-ng#2608] [General]
hello, i'm trying to run airmon-ng start wlan0 on my system and this is the output :

ls: cannot access '/sys/class/net': No such file or directory

and i have nethunter in termux on android also there is no /sys/class directory and on working kali linux there  is all network interfaces in this folder /sys/class/net or drivers and im guessing that why it doesn't works i dont have wlan0 folder in there.

so i'm wondering what is in that /sys/class/net/wlan0 folder and how can i get that so comand ( airmon-ng start wlan0) works

**Comment by ZeroChaos-** [comment]
This is long fixed in git, feel free to use the latest from git.

**Reply by demetrebadzaradze** [reply]
i cant find anything

**Comment by ZeroChaos-** [comment]
"git" refers to the unreleased development version (which is tracked in a tool called git) and available here https://github.com/aircrack-ng/aircrack-ng
You can grab scripts/airmon-ng.linux from there and it should work fine.

**Reply by demetrebadzaradze** [reply]
so you are saying that is going to work even if i dont have this directory /sys/class/net/wlan0 ? and sys directory is empty to

**Comment by ZeroChaos-** [comment]
Probably not, but it won't error :-)

**Comment by kimocoder** [comment]
I see this issue on newer NETLINK / BPF systems, like OpenWRT devices with fresh kernels we had this issue.

Me and ZeroBeat / ZerBea (rather him working, me supplying the device) not too long ago.

https://github.com/ZerBea/hcxdumptool/commit/1f018b8e1d4187fccb297b3481064cb196230ce7
https://github.com/ZerBea/hcxdumptool/commit/82188163eb9fc0da75ec2a11f05414e197a11ba6

And the commits around that soecific

**Reply by ZeroChaos-** [reply]
This doesn't look related to sysfs at all?

**Comment by kimocoder** [comment]
There has been big changes to NETLINK, and it was the indexing of phy0, I see it on several devices which runs newer kernels and was related to that :)

**Reply by ZeroChaos-** [reply]
Cool story. Netlink isn't sysfs and airmon-ng is a shell script accessing sysfs.  Please don't hijack other people's bugs with unrelated issues.

**Comment by kimocoder** [comment]
Pardon? well, not my intention to hijack anything, I just see this issue now and then, sharing my experience causing it. Have a wonderful week 👍
